### PR TITLE
Refactor huggingface and rolling batch handlers

### PR DESCRIPTION
**Pull request by sindhuvahinis**
### Testing

Tested in my EC2 local machine. Built a container and ran the CI by pushing the image in my private dockerhub account. 
https://github.com/deepjavalibrary/djl-serving/actions/runs/9019241325/job/24781785296

In the CI, one of the test is failing, but can't reproduce it in my local EC2 machine.

### CODE REVIEW

File: engines/python/setup/djl_python/huggingface.py [hunk 1]
```diff
@@ -172,8 +150,7 @@ def initialize(self, properties: dict):
                 self.load_stopping_criteria_list(properties["stop_sequence"])
 
         self.input_format_configs = InputFormatConfigs(
-            is_rolling_batch=is_rolling_batch_enabled(
-                self.hf_configs.rolling_batch),
+            is_rolling_batch=False,
```
Comment (frankfliu) [inline engines/python/setup/djl_python/huggingface.py:153]: Why this always false?

File: engines/python/setup/djl_python/huggingface.py [hunk 2]
```diff
@@ -593,9 +497,14 @@ def handle(inputs: Input):
...
     Default handler function
     """
+    global _service
     if not _service.initialized:
+        properties = inputs.get_properties()
+        if "rolling_batch" in properties and properties.get(
+                "rolling_batch") != RollingBatchEnum.disable.value:
+            _service = RollingBatchService()
         # stateful model
-        _service.initialize(inputs.get_properties())
```
Comment (frankfliu) [inline engines/python/setup/djl_python/huggingface.py:598]: Basically, you made the RollingBatch class a top level entryPoint
Let's have a design discussion about this decision.

Comment (sindhuvahinis) [reply engines/python/setup/djl_python/huggingface.py:598]: Sure. Not really a top level entryPoint. RollinBatch does not have `handle` method in it. And the UX still does not change, entryPoint would still be `djl_python.huggingface` for all the vllm, lmi-dist, scheduler rolling batch entryPoint.

File: engines/python/setup/djl_python/rolling_batch/rolling_batch_service.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    "GPTNeoXForCausalLM",
+    "T5ForConditionalGeneration",
+    "LlamaForCausalLM",
+    "FalconForCausalLM",
+    "MPTForCausalLM",
+    "GPTBigCodeForCausalLM",
+}
+
+
+def get_rolling_batch_class_from_str(rolling_batch_type: str, is_mpi: bool,
```
Comment (lanking520) [inline engines/python/setup/djl_python/rolling_batch/rolling_batch_service.py:29]: does TRTLLM also falls into this category?

Comment (sindhuvahinis) [reply engines/python/setup/djl_python/rolling_batch/rolling_batch_service.py:29]: This PR does not cover it. But it should be easy to re-use this code for trtllm as well.

**Pull request closed without merging**